core/src/main/java/hudson/model/ManagementLink.java

```diff
@@ -214,5 +214,26 @@ public enum Category {
         }
     }
 
+    /**
+     * The content of a badge shown as overlay over the icon on "Manage Jenkins".
+     *
+     * @return badge text or {@code null} if no badge should be shown.
+     * @since TODO
+     */
+    public @CheckForNull String getBadge() {
+        return null;
+    }
```

Comment:
This should probably be limited to returning numbers, or is there a real use case for non-numeric badge values that wouldn't quickly overwhelm the badge size?

Reply:
I was thinking for my agent maintenance plugin to display something like 5/12 with a tooltip "5 active maintenance windows 12 total"

Reply:
Another use case might be to show a "!" to indicate that some action needs to be taken

Reply:
Interesting, good point with 5/12.

Still, I wonder whether we could define this API so that it prevents weird abuse (like actual text, as happened in `groovy-postbuild` with build badges)

Reply:
when returning 1000000 it will probably also look weird. One could also think of an array if integers and take only the first 2 or 3 elements of the array and separate it with "/".
Or just take a substring of the first 8 to 10 characters.

Reply:
Some samples how it would look:

Long string without blanks
![image](https://user-images.githubusercontent.com/17767050/184668112-85340587-29d1-4ec4-97b4-13b4cff29e46.png)

String with blanks leads to a line break
![image](https://user-images.githubusercontent.com/17767050/184668200-4f703b07-3ccb-4175-8abf-952e33bc49f5.png)

![image](https://user-images.githubusercontent.com/17767050/184668334-082ec05a-e18e-4210-bc8c-59cd6228aa6a.png)

![image](https://user-images.githubusercontent.com/17767050/184668582-da42070e-65c2-4550-82e6-60e30efdacdb.png)

Reply:
> when returning 1000000 it will probably also look weird.

Realistically, no number will grow that large. The upper bounds should be hundreds of plugins need updating, (tens of) thousands of jobs fail to load, so that's ~5 digits?

Reply:
Also, if it's a string you'll need to define whether it supports HTML or not. Right now it would (and will need to escape any user content).